src/components/hero-carousel/hero-carousel.jsx

```diff
@@ -0,0 +1,52 @@
+import React from "react";
+
+export default React.createClass({
+  propTypes: {
+    contents: React.PropTypes.arrayOf(React.PropTypes.shape({
+      image: React.PropTypes.string.isRequired,
+      caption: React.PropTypes.string.isRequired,
+      attribution: React.PropTypes.string.isRequired
+    })).isRequired
+  },
+  getInitialState() {
+    return {
+      activeTrigger: 0
+    };
+  },
+  onTriggerClick: function (index) {
+    this.setState({
+      activeTrigger: index
+    });
+  },
+  render: function() {
+    let triggers = this.props.contents.map((item, index) => {
+      let classNames = `hero${this.state.activeTrigger === index ? ` active` : ``}`;
+
+      return (
+        <button key={index} className={classNames} onClick={this.onTriggerClick.bind(null, index)}>
```

Comment:
relying on syntax rather than low level JS functions, mainly. The arrow notation does "out of the box" what this low level spec function call with null-context argument is used for, so it's going to be easier to understand (once you're familiar with it, it's easy to forget that it's a pretty hardcore "you should never need this in normal code" function)